### ISSUE TITLE
Bug when not on 100% zoom

### ISSUE THREAD

**Issue by eduardomourar** [jsGanttImproved/jsgantt-improved#23]
If the browser is viewing the page at anything other than 100% zoom then bars may not be sized or positioned correctly.

**Comment by mahmad2504** [comment]
Even with 100% zoom i am experiencing this issue. Tasks gets offset by weeks in large duration projects

https://user-images.githubusercontent.com/15646324/28788779-d048ded6-763b-11e7-9e4d-04643777bcc0.png

**Comment by RickCardoso** [comment]
This seems to be the same issue as #36 and #43 which are being fixed in the pull request #41. I'll also close it, leaving #36 open and avoiding multiplicity.

**Comment by eduardomourar** [comment]
PR #41 fixes partially this issue, but I will keep it open for future reference.

**Comment by mariohmol** [comment]
tested in v2 working pretty good

**Comment by nidu** [comment]
I have issues with zooming in current version as well. For example here's the demo page with 125% zoom in Chrome:

![image](https://user-images.githubusercontent.com/434377/62696859-cd5a9280-b9f2-11e9-9ec5-e80899477c40.png)

**Comment by mariohmol** [comment]
thats true.. table may not be very good for this.. nowdays divs are best suit for design right?

but i guess that proper css (using % maybe) shuld work with tables.. no?

but thats a good idea  to try out divs instead

**Comment by nidu** [comment]
Probably CSS will do, but I can't find anything on it, no related properties or methods. Probably i'm missing something.

The only way i can imagine solving it with tables at the moment is to do bar positioning based on cell absolute offsets after headers is added, which sounds like quite a mess.

**Comment by nidu** [comment]
@mariohmol shouldn't this one be reopened? It's easily reproducible at the moment on the demo page.

**Comment by nidu** [comment]
I'm working on this issue and converted chart to divs. With divs it works quite fine.

Here's current demo at 100%. I'm not sure why but task end dates look a bit misplaced to the right.
![image](https://user-images.githubusercontent.com/434377/74060619-a6553b80-49ea-11ea-8c15-025db7de0a34.png)

Here's it at 80%:
![image](https://user-images.githubusercontent.com/434377/74060688-cc7adb80-49ea-11ea-9011-54200ed5db7c.png)

Here's div version at 100%:
![image](https://user-images.githubusercontent.com/434377/74060730-e1576f00-49ea-11ea-88db-314724d2441f.png)

And at 80%:
![image](https://user-images.githubusercontent.com/434377/74060761-efa58b00-49ea-11ea-9c9a-168030907b33.png)

So the bars are properly placed now, but
- Some borders disappear
- Chart doesn't match list anymore. Might be table scaling issues again, so looks like if i convert chart to divs - task list must be converted too

What i don't like about it is that this is a breaking change in a lot of use cases.

I'm still looking into other options.

Source is here https://github.com/nidu/jsgantt-improved/tree/23-div-based-tables

**Comment by nidu** [comment]
Ok it seems like there's a better solution.

If i just set table-layout:fixed for chart tables and set widths for td - the issues is fixed. For example here are again screenshots of current version (top) and new version (bottom) at 80% zoom:
![image](https://user-images.githubusercontent.com/434377/74073978-93536300-4a0c-11ea-8981-9968b2692d0a.png)

This change is much simpler and less breaking. As a bonus according to [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/table-layout) fixed table layout is faster.

Here's the [change](https://github.com/nidu/jsgantt-improved/commit/43b35ed177af4d90f3f731eab92185143c716836).

@mariohmol do you think we can go with this approach? I don't see any issues with it so far

**Comment by mariohmol** [comment]
this looks good.. as we keep the same HTML structure and change basically CSS..

if this works good when u test.. put in a PR and lets use this =)

**Comment by mariohmol** [comment]
## Status update — fix verified ✅

We investigated this today and the `table-layout: fixed` fix originally proposed in this thread is already in place in the current codebase (`jsgantt.css` lines 219–221):

```css
.gcharttable,
.gcharttableh {
  table-layout: fixed;
}
```

We added a Playwright e2e test suite to formally verify this (PR #392). The test applies CSS zoom at 75%, 80%, 100%, 125%, and 150% and asserts:

1. **Bar alignment** — each bar's rendered position scales proportionally with zoom (within ±3px)
2. **Column uniformity** — all columns within a zoom level have the same width (spread ≤1px), confirming no fractional-px accumulation from `table-layout: auto`

All 10 test cases pass. The issue appears to be resolved in the current version.
